src/inspector_agent.cc

```diff
@@ -207,6 +207,7 @@ class AgentImpl {
   bool wait_;
   bool connected_;
   bool shutting_down_;
+  bool startup_failed_;
```

Comment:
Thank you for the review. I introduced initial set of states. Note that I am still working a refactoring that will split "inspector agent" (the one that lives on the main thread and talks to V8) and "inspector server" (that talks to a client over the socket - including all HTTP responses and such). Then there will be 2 objects with likely two states. In the current implementation, adding states for all flags (like shutting_down_ and wait_) would make the variable accessed  (and modified) from both threads - which might introduce some race conditions (or require more synchronization than is neccessary now).